### ISSUE TITLE
Companion does not change contents of /etc/nginx/conf.d/default.conf

### ISSUE THREAD

**Issue by ziglef** [nginx-proxy/docker-letsencrypt-nginx-proxy-companion#684]
As the title states, when I run the container it won't change the contents of the config file for NGINX thus making it unable to handle the challenges.

My setup consists of a docker-compose file:
```
version: "3"

services:
  web:
    build:
      context: ../backend/src
      dockerfile: prod.Dockerfile
    volumes:
      - ../backend/src/:/usr/src/app/:delegated
    environment:
      - VIRTUAL_HOST=example-host.site.com
      - VIRTUAL_PORT=8000
      - LETSENCRYPT_HOST=example-host.site.com
      - LETSENCRYPT_EMAIL=example@site.com
      - LETSENCRYPT_RESTART_CONTAINER=true
      - DATABASE_URL=some_url
      - DATABASE_POOL_SIZE=some_int
    depends_on:
      - letsencrypt
      - nginx-proxy
    restart: always

  nginx-proxy:
    image: jwilder/nginx-proxy
    container_name: nginx-proxy
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - conf:/etc/nginx/conf.d
      - ./example-host.site.com:/etc/nginx/vhost.d/example-host.site.com:ro
      - vhost:/etc/nginx/vhost.d
      - html:/usr/share/nginx/html
      - dhparam:/etc/nginx/dhparam
      - certs:/etc/nginx/certs:ro
      - /var/run/docker.sock:/tmp/docker.sock:ro
    restart: always

  letsencrypt:
    image: jrcs/letsencrypt-nginx-proxy-companion
    environment:
      - NGINX_PROXY_CONTAINER=nginx-proxy
      - DEBUG=true
      - ACME_CA_URI=https://acme-staging-v02.api.letsencrypt.org/directory
    volumes:
      - conf:/etc/nginx/conf.d
      - vhost:/etc/nginx/vhost.d
      - html:/usr/share/nginx/html
      - dhparam:/etc/nginx/dhparam
      - certs:/etc/nginx/certs:rw
      - /var/run/docker.sock:/var/run/docker.sock:ro
    depends_on:
      - nginx-proxy
    restart: always

volumes:
  conf:
  vhost:
  html:
  dhparam:
  certs:

networks:
  default:
    external:
      name: nginx-proxy
```

Error happens even when not using DEBUG or ACME_CA_URI.

Log follows:
```
letsencrypt_1  | Generating a RSA private key
letsencrypt_1  | .....................................................................................++++
letsencrypt_1  | .......++++
letsencrypt_1  | writing new private key to '/etc/nginx/certs/default.key.new'
letsencrypt_1  | -----
letsencrypt_1  | Info: a default key and certificate have been created at /etc/nginx/certs/default.key and /etc/nginx/certs/default.crt.
letsencrypt_1  | Debug: checking /etc/nginx/certs/default.key ownership and permissions.
letsencrypt_1  | Debug: numeric ID of user root is 0.
letsencrypt_1  | Debug: numeric ID of group root is 0.
letsencrypt_1  | Debug: setting /etc/nginx/certs/default.key permissions to 644.
letsencrypt_1  | Debug: checking /etc/nginx/certs/default.crt ownership and permissions.
letsencrypt_1  | Debug: numeric ID of user root is 0.
letsencrypt_1  | Debug: numeric ID of group root is 0.
letsencrypt_1  | Info: Creating Diffie-Hellman group in the background.
letsencrypt_1  | A pre-generated Diffie-Hellman group will be used for now while the new one
letsencrypt_1  | is being created.
letsencrypt_1  | Debug: checking /etc/nginx/certs/dhparam.pem ownership and permissions.
letsencrypt_1  | Debug: numeric ID of user root is 0.
letsencrypt_1  | Debug: numeric ID of group root is 0.
letsencrypt_1  | Generating DH parameters, 2048 bit long safe prime, generator 2
letsencrypt_1  | Reloading nginx proxy (nginx-proxy)...
letsencrypt_1  | 2020/07/28 13:38:33 Generated '/etc/nginx/conf.d/default.conf' from 3 containers
letsencrypt_1  | 2020/07/28 13:38:33 [notice] 72#72: signal process started
letsencrypt_1  | 2020/07/28 13:38:33 Generated '/app/letsencrypt_service_data' from 3 containers
letsencrypt_1  | 2020/07/28 13:38:33 Running '/app/signal_le_service'
letsencrypt_1  | 2020/07/28 13:38:33 Watching docker events
letsencrypt_1  | 2020/07/28 13:38:33 Contents of /app/letsencrypt_service_data did not change. Skipping notification '/app/signal_le_service'
letsencrypt_1  | /etc/nginx/certs/_test_example-host.site.com /app
letsencrypt_1  | Reloading nginx proxy (nginx-proxy)...
letsencrypt_1  | 2020/07/28 13:38:35 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification ''
letsencrypt_1  | 2020/07/28 13:38:35 [notice] 101#101: signal process started
letsencrypt_1  | Creating/renewal example-host.site.com certificates... (example-host.site.com)
letsencrypt_1  | 2020-07-28 13:38:37,451:INFO:simp_le:1359: Generating new account key
letsencrypt_1  | 2020-07-28 13:38:39,923:INFO:simp_le:1387: By using simp_le, you implicitly agree to the CA's terms of service: https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf
letsencrypt_1  | 2020-07-28 13:38:40,249:INFO:simp_le:1450: Generating new certificate private key
letsencrypt_1  | 2020-07-28 13:38:46,925:ERROR:simp_le:1417: CA marked some of the authorizations as invalid, which likely means it could not access http://example.com/.well-known/acme-challenge/X. Did you set correct path in -d example.com:path or --default_root? Are all your domains accessible from the internet? Please check your domains' DNS entries, your host's network/firewall setup and your webserver config. If a domain's DNS entry has both A and AAAA fields set up, some CAs such as Let's Encrypt will perform the challenge validation over IPv6. If your DNS provider does not answer correctly to CAA records request, Let's Encrypt won't issue a certificate for your domain (see https://letsencrypt.org/docs/caa/). Failing authorizations: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/84311592
letsencrypt_1  | 2020-07-28 13:38:46,929:INFO:simp_le:401: Saving account_key.json
letsencrypt_1  | 2020-07-28 13:38:46,933:INFO:simp_le:401: Saving account_reg.json
letsencrypt_1  | Challenge validation has failed, see error log.
letsencrypt_1  |
letsencrypt_1  | Debugging tips: -v improves output verbosity. Help is available under --help.
letsencrypt_1  | /app
letsencrypt_1  | Sleep for 3600s
letsencrypt_1  | This is going to take a long time
letsencrypt_1  | Info: Diffie-Hellman group creation complete, reloading nginx.
letsencrypt_1  | Debug: checking /etc/nginx/certs/dhparam.pem ownership and permissions.
letsencrypt_1  | Debug: numeric ID of user root is 0.
letsencrypt_1  | Debug: numeric ID of group root is 0.
letsencrypt_1  | Reloading nginx proxy (nginx-proxy)...
letsencrypt_1  | 2020/07/28 13:39:05 Contents of /etc/nginx/conf.d/default.conf did not change. Skipping notification ''
letsencrypt_1  | 2020/07/28 13:39:05 [notice] 130#130: signal process started
```

What seems to be happening by analysing the last modified times of the volume folders is that somehow the companion container can't change the /etc/nginx/conf.d/default.conf file.

This happens on AWS using Amazon Linux 2 AMI 2.0.20200617.0 x86_64 HVM as a base image for the VM on EC2
Weirdly the same file works on GCP using their own base image.

While this could be an OS level issue, I'm not sure what could / should be changed, and if something can be done here.

**EDIT**: NVM the AWS and GCP stuff.
It was working on GCP because my image of the companion was older, unfortunately I did not record which image version I was using before.
After nuking the install and using :latest it stopped working, exhibiting the same behaviour as AWS
Maybe something recent broke this. I will dig into the change log and try to figure it out

**Comment by buchdag** [comment]
> somehow the companion container can't change the `/etc/nginx/conf.d/default.conf` file.

The companion does not touch this file at all, it is generated by the `docker-gen` process running inside your `nginx-proxy` container.

Amazon AMI, at least the one they use on ECS, has been known to cause issues with `docker-gen` : see #453

**Comment by ziglef** [comment]
I just tested the config on Google Cloud and the same issue happens, so I don't think this is exclusive to AWS. I also don't use ECS, but maybe the cause happens the same on EC2 alone.

I will check `docker-gen` and `nginx-proxy` to see if I can isolate the issue.

If I come back to this maybe being a companion issue I will re-open.

Closing for now as it doesn't seem related to companion.

Thanks @buchdag
